app/src/main/java/de/westnordost/streetcomplete/quests/building_entrance_reference/AddEntranceReference.kt

```diff
@@ -0,0 +1,101 @@
+package de.westnordost.streetcomplete.quests.crossing
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.quest.NoCountriesExcept
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BLIND
+import de.westnordost.streetcomplete.quests.building_entrance_reference.AddEntranceReferenceForm
+import de.westnordost.streetcomplete.quests.building_entrance_reference.EntranceAnswer
+import de.westnordost.streetcomplete.quests.building_entrance_reference.FlatRange
+import de.westnordost.streetcomplete.quests.building_entrance_reference.ReferenceCode
+import de.westnordost.streetcomplete.quests.building_entrance_reference.ReferenceCodeAndFlatRange
+import de.westnordost.streetcomplete.quests.building_entrance_reference.Unsigned
+
+class AddEntranceReference : OsmElementQuestType<EntranceAnswer> {
+
+    private val buildingFilter by lazy { """
+        ways with
+          building = apartments
+          and ruins != yes and disused != yes
```

Comment:
Does it make sense to require/exclude either/any of:
- `addr:housenumber` on the apartment building
- `addr:housenumber` on the entrance

?

Reply:
addr:housenumber on the apartment building - no, code on entrance is not replacement in such case for housenumber address

addr:housenumber on the entrance - I am not 100% sure, as while it is not excluding in principle then it would indicate that someone mapped this or relatively rare case in this countries where each entrance has own house number